deploy/cloudformation/gomain.go

```diff
@@ -34,10 +34,21 @@ import (
 )
 
 const (
-	prodTemplatePath = "elastic-agent-ec2.yml"
-	devTemplatePath  = "elastic-agent-ec2-dev.yml"
+	DEV  = "DEV"
+	PROD = "PROD"
 )
 
+var templatePaths = map[string]map[string]string{
+	DeploymentTypeCSPM: map[string]string{
+		DEV:  "elastic-agent-ec2-dev-cspm.yml",
+		PROD: "elastic-agent-ec2-cspm.yml",
+	},
+	DeploymentTypeCNVM: map[string]string{
+		DEV:  "elastic-agent-ec2-dev-cnvm.yml",
+		PROD: "elastic-agent-ec2-cnvm.yml",
+	},
+}
```

Comment:
nit: Create a struct
```
type flavor struct {
    Name string
    DevTemplate string
    ProdTemplate string
}
```

Reply:
Discussed together and came to conclusion that we would like to keep the map access dynamic for now.
We can re-evaluate in future if needed.